routing/single_mwm_router.cpp

```diff
@@ -200,39 +201,26 @@ bool SingleMwmRouter::LoadIndex(MwmSet::MwmId const & mwmId, string const & coun
   }
 }
 
-bool SingleMwmRouter::BuildRoute(MwmSet::MwmId const & mwmId, vector<Joint::Id> const & joints,
+bool SingleMwmRouter::BuildRoute(MwmSet::MwmId const & mwmId, vector<Segment> const & segments,
                                  RouterDelegate const & delegate, IndexGraphStarter & starter,
                                  Route & route) const
 {
-  vector<RoutePoint> routePoints;
-  starter.RedressRoute(joints, routePoints);
-
-  // ReconstructRoute removes equal points: do it self to match time indexes.
-  // TODO: rework ReconstructRoute and remove all time indexes stuff.
-  routePoints.erase(unique(routePoints.begin(), routePoints.end(),
-                           [&](RoutePoint const & rp0, RoutePoint const & rp1) {
-                             return starter.GetPoint(rp0.GetRoadPoint()) ==
-                                    starter.GetPoint(rp1.GetRoadPoint());
-                           }),
-                    routePoints.end());
-
   vector<Junction> junctions;
-  junctions.reserve(routePoints.size());
+  junctions.reserve(segments.size());
```

Comment:
Am I right that you should use GetRouteNumPoints() here instead of segments.size()?